### PR TITLE
feat: add allowlist filter mode for rsn advanced config

### LINKED ISSUE #305: Reopen discussion on rsn whitelist option
I would once again like to request a rsn whitelist feature.

I am the original requester of this feature (https://github.com/pajlads/DinkPlugin/issues/111), and even though a whitelist suited my use case at the time. Over the last year multiple friends have expressed disappointment of not being able to configure a whitelist. For some friends this was a deal breaker and they chose to stick to other (older/abanonded) plugins even they prefer the features of Dink. Therefore I now believe the user should be able to choose between a whitelist and an ignorelist.

I understand it could be confusing for some users and therefore propose the following:

A menu called 'Filter mode' that allows the user to choose between 'Allow all RSNs' and 'Ignore all RSNs', this option would sit above the current input box for RSNs. This way of doing is is not unheard of in runelite plugins, for example the [Clan Spam Filter](https://github.com/andreasx23/GuildSpamFilter) plugin has a similar toggle:

JLGyomp

The current 'Ignored RSN's' input box would also be renamed to a more apt 'RSN list' or 'List of RSNs' input box, as well as the tooltip.
In addition the default should still be set to 'Ignore all RSNs' so that the plugin doesn't break if a user fails to spot the configuration option.

Happy to go back on forth on this if needed, it really is a feature me and my friend group would love to see.

Thanks!

### PR DESCRIPTION
**Pull request by iProdigy**
Closes #305

Uses @Felanbird's suggestion of a single input box and allow/deny toggle in https://github.com/pajlads/DinkPlugin/pull/114#issuecomment-1375997995 to avoid complexity of the initial approach of having a separate box for the allowlist and denylist.

Still, we discourage use of the allow filter mode since users can get confused when a new account does not fire notifications (because they have not added it to the input box).

config

### CODE REVIEW

**Comment by Newbcake** [comment]
Hey just wanna show my thanks here, seriously appreciate it.

File: src/main/java/dinkplugin/DinkPluginConfig.java [hunk 1]
```diff
@@ -241,21 +242,34 @@ default String embedFooterIcon() {
     }
 
     @ConfigItem(
-        keyName = "ignoredNames",
-        name = "Ignored RSNs",
-        description = "Prevent notifications from the following player names (One name per line)",
+        keyName = "ignoredNames", // historical name, preserved for backwards compatibility
+        name = "Filtered RSNs",
```
Comment (pajlada) [inline src/main/java/dinkplugin/DinkPluginConfig.java:246]: Name choice is good

**Review by pajlada — APPROVED** [review]
Everything looks good to me, with sane defaults & the phrasing we use I think this is good to merge :+1: 
A language-check on the README/changelog and it should be good to go

**Comment by iProdigy** [comment]
A remaining complication: on config import, the current settings and the imported settings may have a different name filter mode. Seems weird to combine the text boxes when the modes differ?

File: src/main/java/dinkplugin/DinkPluginConfig.java [hunk 2]
```diff
@@ -241,21 +242,34 @@ default String embedFooterIcon() {
...
     )
-    default String ignoredNames() {
+    default String filteredNames() {
         return "";
     }
 
+    @ConfigItem(
+        keyName = "nameFilterMode",
+        name = "RSN Filter Mode",
+        description = "Allow Mode: Only allow notifications for RSNs on the list above.<br/>" +
```
Comment (iProdigy) [inline src/main/java/dinkplugin/DinkPluginConfig.java:259]:
```suggestion
        description = "Allow Mode: Only allow notifications for RSNs on the list above (discouraged).<br/>" +
```

Do we want to add a warning here?

**Comment by iProdigy** [comment]
@Felanbird could you also review the wording of settings/readme please

File: src/main/java/dinkplugin/SettingsManager.java [hunk 3]
```diff
@@ -83,19 +84,21 @@ public class SettingsManager {
     private final ConfigManager configManager;
 
     /**
-     * Check whether a username is not on the configured ignore list.
+     * Check whether a username adheres to the configured RSN filter list.
```
Comment (Felanbird) [inline src/main/java/dinkplugin/SettingsManager.java:87]:
```suggestion
     * Check whether a username complies with the configured RSN filter list.
```

I think adhere basically never fits in coding, also comply was already used in the readme

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,6 @@
 ## Unreleased
 
+- Minor: Add allowlist mode for RSN-based notification filter under the Advanced category. Use this setting with caution. (#306)
```
Comment (Felanbird) [inline CHANGELOG.md:3]:
```suggestion
- Minor: Add allowlist mode for Filtered RSNs under the Advanced category. Use this setting with caution. (#306)
```

**Review by Felanbird — CHANGES_REQUESTED** [review]

**Review by Felanbird — APPROVED** [review]

**Pull request merged**